internal/runtime/hcsv2/workload_container.go

```diff
@@ -16,6 +18,40 @@ func getWorkloadRootDir(sbid, id string) string {
 	return filepath.Join(getSandboxRootDir(sbid), id)
 }
 
+func createSandboxMountPath(sandboxSource string) error {
+	parentPath, _ := filepath.Split(sandboxSource)
+	pathExt := filepath.Ext(sandboxSource)
+
+	if err := os.MkdirAll(parentPath, 0755); err != nil {
+		return err
+	}
+
+	if pathExt == "" {
+		return os.Mkdir(sandboxSource, 0755)
+	}
+	_, err := os.Create(sandboxSource)
+	return err
+}
+
+func updateSandboxMounts(sbid string, spec *oci.Spec) error {
+	sandboxMountPrefix := "sandbox://"
+	for i, m := range spec.Mounts {
+		if strings.HasPrefix(m.Source, sandboxMountPrefix) {
+			subPath := strings.TrimPrefix(m.Source, sandboxMountPrefix)
+			sandboxSource := filepath.Join(getSandboxMountsDir(sbid), subPath)
+			spec.Mounts[i].Source = sandboxSource
+
+			_, err := os.Stat(sandboxSource)
+			if os.IsNotExist(err) {
```

Comment:
I dont think we should handle creating a file. If we do a `os.Stat` on sandbox source we should just issue a mkdir all and move on. If the stat comes back as valid then its either a file or directory within the sandbox mount and is already a valid source.

Reply:
Done